rotkehlchen/chain/optimism/decoding/decoder.py

```diff
@@ -48,7 +49,21 @@ def __init__(
 
     # -- methods that need to be implemented by child classes --
 
-    def _enrich_protocol_tranfers(self, context: 'EnricherContext') -> TransferEnrichmentOutput:  # pylint: disable=unused-argument
+    def _enrich_protocol_tranfers(self, context: 'EnricherContext') -> TransferEnrichmentOutput:
+        for enrich_call in self.rules.token_enricher_rules:
```

Comment:
you are repeating code... Does it need to exists for all the chains? What happens for example with scroll? If so instead of modifying it overwrite the method in the evm class. If not you can avoid calling the logic by modifying `_enrich_protocol_tranfers` in the base class and having something like

```python
if not self.should_enrich_transfers:
     return Failed_enrichment output

_this_logic_that_you_repeat_
```